internal/provider/iam/resource_group.go

```diff
@@ -0,0 +1,225 @@
+package iam
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-iam/stable/2019-12-10/client/groups_service"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-iam/stable/2019-12-10/models"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	clients "github.com/hashicorp/terraform-provider-hcp/internal/clients"
+	"github.com/hashicorp/terraform-provider-hcp/internal/hcpvalidator"
+)
+
+func NewGroupResource() resource.Resource {
+	return &resourceGroup{}
+}
+
+type resourceGroup struct {
+	client *clients.Client
+}
+
+func (r *resourceGroup) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_group"
+}
+
+func (r *resourceGroup) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: fmt.Sprintf(`The group resource manages a HCP Group.
+
+The user or service account that is running Terraform when creating a %s resource must have %s on the parent resource; either the project or organization.`,
+			"`hcp_group`", "`roles/admin`"),
+		Attributes: map[string]schema.Attribute{
+			"resource_id": schema.StringAttribute{
+				Computed:    true,
+				Description: "The group's unique identifier",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"resource_name": schema.StringAttribute{
+				Computed: true,
+				Description: fmt.Sprintf("The group's resource name in the format `%s` or `%s`",
+					"iam/organization/<organization_id>/service-principal/<name>", "<name>"),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"display_name": schema.StringAttribute{
+				Required:    true,
+				Description: "The group's display_name.",
+				Validators: []validator.String{
+					hcpvalidator.ResourceNamePart(),
+					// TODO: we don't enforce a list for the display name in the portal, do we care here?
```

Comment:
We might be able to follow the validator from `cloud-iam`: https://github.com/hashicorp/cloud-iam/blob/main/service/private/groups/service_create_group.go#L87

The link points to Name so you might be correct about not having a limit for the Display Name.

I guess that also brings up the question: do we _want_ to have a limit on the length?

Reply:
Ahh nice catch, thanks for linking to that bit of code @insanejasper. I updated the description length as suggested, and I think it's safe to restrict the display name max length to 90 perhaps? The resource name max length is 125 and since the resource name is generated from the display name, it should be enforced as shorter to leave space for the `iam/organization/<org-id>/group` prefix. Just pushed up changes to reflect this, and we also can discuss further if needed!

Reply:
Do we need to leave room for a nonce if there is a collision with another resource name?

Reply:
Hmm, I agree that we should yes. What length should we expect a given nonce to be @insanejasper? Would the length be set or variable?

Reply:
Just saw the PR go up, and it looks like the nonce is a UUID (correct me if I'm wrong) of about 35 characters. That said, an upper restriction of 50 characters would be reasonable for the display name I think to account for nonce length.

Reply:
That sounds good to me. I did ask for some feedback on the nonce length but if anything it'll be shorter if it does change!